### DISCUSSION TITLE
--include and --exclude does not work together

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [RsyncProject/rsync#292] [General]
I'm trying to sync files with pattern `*/ls` from folder /bin to folder test:

```
mkdir test
rsync -avv --exclude='*' --include=*/ls /bin/ test/
ls test
```

And from the output the include filter failed to work:

```
[sender] hiding file ls because of pattern *
```

Anyone know what's wrong?

**Comment by CaledoniaProject** [comment]
Resolved, need an --include='*/' before the exclusion filter

**Comment by WayneD** [comment]
Note also that option order is important.  The very newest man page online has a little better description about this:

https://download.samba.org/pub/rsync/rsync.1#FILTER_RULES
